packages/components/src/guide/types.ts

```diff
@@ -0,0 +1,64 @@
+/**
+ * External dependencies
+ */
+import type { ReactNode } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import type { ModalProps } from '../modal/types';
+
+export type Page = {
+	/**
+	 * Content of the page.
+	 */
+	content: ReactNode;
+	/**
+	 * Image displayed above the page content.
+	 */
+	image?: ReactNode;
+};
+
+export type GuideProps = {
+	/**
+	 * Deprecated. Use `pages` prop instead.
+	 *
+	 * @deprecated since 5.5
+	 */
+	children?: ReactNode;
+	/**
+	 * A custom class to add to the modal.
+	 */
+	className?: string;
+	/**
+	 * Used as the modal's accessibility label.
+	 */
+	contentLabel: ModalProps[ 'contentLabel' ];
+	/**
+	 * Use this to customize the label of the _Finish_ button shown at the end of the guide.
+	 */
+	finishButtonText?: string;
+	/**
+	 * A function which is called when the guide is finished.
+	 */
+	onFinish: ModalProps[ 'onRequestClose' ];
+	/**
+	 * An array of objects describing each page in the guide.
+	 */
+	pages?: Page[];
```

Comment:
Missing default prop value:

```suggestion
	/**
	 * An array of objects describing each page in the guide.
	 *
	 * @default []
	 */
	pages?: Page[];
```

Reply:
Fixed via https://github.com/WordPress/gutenberg/pull/47493/commits/b61c545d515c109022a41d16a86a3c1fda365d5d 👍🏻